src/Analysis/Ast/Test/InheritNonClassTests.cs

```diff
@@ -0,0 +1,161 @@
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Python.Analysis.Diagnostics;
+using Microsoft.Python.Analysis.Tests.FluentAssertions;
+using Microsoft.Python.Core;
+using Microsoft.Python.Parsing.Tests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestUtilities;
+
+namespace Microsoft.Python.Analysis.Tests {
+    [TestClass]
+    public class InheritNonClassTests : AnalysisTestBase {
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+            => TestEnvironmentImpl.TestInitialize($"{TestContext.FullyQualifiedTestClassName}.{TestContext.TestName}");
+
+        [TestCleanup]
+        public void Cleanup() => TestEnvironmentImpl.TestCleanup();
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromRenamedBuiltin() {
+            const string code = @"
+tmp = str
+
+class C(tmp):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(0);
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromBuiltin() {
+            const string code = @"
+class C(str):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(0);
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromUserClass() {
+            const string code = @"
+class D:
+    def hello(self):
+        pass
+
+class C(D):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(0);
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromConstant() {
+            const string code = @"
+class C(5):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(1);
+
+            var diagnostic = analysis.Diagnostics.ElementAt(0);
+            diagnostic.SourceSpan.Should().Be(2, 7, 2, 8);
+            diagnostic.Message.Should().Be(Resources.InheritNonClass.FormatInvariant("5"));
+            diagnostic.ErrorCode.Should().Be(ErrorCodes.InheritNonClass);
+        }
+
+
+        [TestMethod, Priority(0)]
+        public async Task InheritFromConstantVar() {
+            const string code = @"
+x = 'str'
+
+class C(x):
+    def method(self):
+        return 'test'
+";
+            var analysis = await GetAnalysisAsync(code);
+            analysis.Diagnostics.Should().HaveCount(1);
+
+            var diagnostic = analysis.Diagnostics.ElementAt(0);
+            diagnostic.SourceSpan.Should().Be(4, 7, 4, 8);
```

Comment:
If you have time, you can add an assertion for the `DiagnosticsEntry` to have better messages in tests.

Reply:
I'm not sure what you mean by this, could you explain more?

Reply:
Take a look, for example, at `CompletionResultAssertions`

Reply:
Oh I see what you mean, so create a framework like ```CompletionResultAssertions``` so that it reads more smoothly!

Reply:
Talked with @CTrando, he'll do this in another PR that adds it and updates all of the other tests to use it.